joomla/joomla-cms
# [5.0] Find proper script for file uninstall

Pull Request for Issue #19085.

### Summary of Changes
When uninstalling an extension of type `file` with a script file, the preflight, postflight and actual uninstall methods are not called. The (un)installer is loading the right manifest and even the right script file, but doesn't look for the right classname.


### Testing Instructions
Get a random extension of type `file` and add an install script by adding `<scriptfile>example.script.php</scriptfile>` to the manifest and for example the attached script file [example.script.php.txt](https://github.com/joomla/joomla-cms/files/9965468/example.script.php.txt) (You have to rename it and remove the .txt at the end. Also make sure that you adapt the scripts class name to your file extensions manifest name.)


### Actual result BEFORE applying this Pull Request
Installing shows `preflighttestpostflight` as install message. Uninstall doesn't show anything.


### Expected result AFTER applying this Pull Request
Both installing and uninstalling shows `preflighttestpostflight` as (un)install message.


### Link to documentations
Please select:
